addons/mrp/models/mrp_workorder.py

```diff
@@ -335,6 +350,12 @@ def _generate_wo_lines(self):
         )
         for move in moves:
             qty_to_consume = self._prepare_component_quantity(move, self.qty_producing)
```

Comment:
_prepare_component_quantity() convert already the quantity depending on the tracking. I don't think you need to do it here

Reply:
It is required.
`_prepare_component_quantity` does not return the correct value when product tracking is `serial`.
The correct result should be computed from 1 unit of the product UOM.
Instead it is computed from 1 unit of the production UOM.
Maybe there is an issue with the `move's unit_factor`.